### ISSUE TITLE
Normalization of 1D plots

### ISSUE THREAD

**Issue by andrewfowlie** [michaelhb/superplot#3]
Hello,

For 1D plots, I've normalized the profile likelihood and posterior pdf so that their maximum values are 1, with something like `pdf = pdf / pdf.max()`

That's OK usually. But I suppose I make two 1D plots of the same quantity in e.g. different models. if I normalize them both so that their maximum values are 1, the comparison isn't quite right. They should be normalized so that their areas are 1.  That way, a broad distribution isn't as high as a narrow distribution. Is it clear?

This should be fixed in the future.

**Comment by michaelhb** [comment]
Makes sense. Do we want to have both options, or is normalizing by area a strictly better way to do it?

**Comment by andrewfowlie** [comment]
I'm not sure. The advantage to normalzing so that max height = 1 is that it's easy to plot profile-likelihood and posterior pdf on the same plot, without one being much bigger than the other.

I don't think it's an issue for the profile likelihood. The true height of the 1D profile-like is determined by the minimum chi-squared, so you'd just compare that between models, and profile-like isn't a pdf anyway. You wouldn't compare prof like between models in the same way as you would a pdf.

Maybe:
- If you plot only posterior, normalize by area
- If you plot profile-like/prof-like and posterior, normalize so that max prof like = max posterior = const, where const = 1 seems like a reasonable choice! Profile like and posterior have different units anyway, so if you put them on the same plot, you can't expect a particularly meaningful y-axis.

**Comment by andrewfowlie** [comment]
Ah, but a potential issue is that the height of things like the credible regions on the 1D plot is just set to a constant (1.2 I think, which puts it above the pdf which has a max at 1) in the plot co-ordinates. They should be set relative to the size of the plot window e.g. 80% of the plot height (i.e. near the top).

**Comment by innisfree** [comment]
On second thoughts, might as well be flexible:

**Comment by michaelhb** [comment]
OK - I think this is working now. Assuming that passing "density=True" to numpy.histogram is doing what we want it to.

I've set it up so that if the prof like is plotted, everything gets normed so the max value is one. If only the PDF is plotted, it gets normed so that the area is one. I set the Y axis to autoscale, and the height of best-fit / posterior mean points, confidence intervals and credible regions is set relative to the maximum data value.

NB only the default 1D plot is affected by these changes - this is what we wanted right?

Cheers!
